app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,67 @@
+class AlbumsController < ApplicationController
+  def index
+    @albums = Album.order(ranking: :desc)
+  end
+
+  def show
+    id = params[:id]
+    @album = Album.find(id)
+  end
+
+  def new
+    @album = Album.new
+  end
+
+  def upvote
+    @album = Album.find(params[:id])
+    if @album.ranking.nil?
+      ranking = 1
+    else
+      ranking = @album.ranking + 1
+    end
+    @album.update(ranking: ranking)
+    redirect_to album_path(@album.id)
+  end
+
+  def create
+    @album = Album.create(album_params[:album])
+    if @album.save
+      redirect_to album_path(@album)
+    else
+      render "new"
+    end
+  end
+
+  def destroy
+    id = params[:id]
+    Album.delete(id)
+    redirect_to albums_path
+  end
+
+  def edit
+    id = params[:id]
+    @album = Album.find(id)
+  end
+
+  def update
+    id = params[:id]
+    @album = Album.find(id)
+    @album.attributes = {
+      name: album_params[:album][:name],
+      artist: album_params[:album][:artist],
+      description: album_params[:album][:description],
```

Comment:
Since you defined a private method album_params, you could've just written:
@album.update(album_params[:album])
But the way you wrote is good too. It's just a good practice to write less lines of code when it's possible